source/common/buffer/buffer_impl.cc

```diff
@@ -118,6 +118,37 @@ void OwnedImpl::copyOut(size_t start, uint64_t size, void* data) const {
   ASSERT(size == 0);
 }
 
+uint64_t OwnedImpl::copyOutToSlices(uint64_t size, Buffer::RawSlice* dest_slices,
+                                    uint64_t num_slice) const {
+  uint64_t total_length_to_read = std::min(size, this->length());
+  uint64_t num_bytes_read = 0;
+  uint64_t num_dest_slices_read = 0;
+  uint64_t num_src_slices_read = 0;
+  uint64_t dest_slice_off = 0;
+  uint64_t src_slice_off = 0;
+  for (; num_dest_slices_read < num_slice && num_bytes_read < total_length_to_read;) {
+    auto& src_slice = slices_[num_src_slices_read];
+    auto& dest_slice = dest_slices[num_dest_slices_read];
+    auto length_to_copy = std::min(
```

Comment:
const, and explicit types please
optional nit: I would be inclined to break this up into a few temporary variables